src/systemctl/systemctl.c

```diff
@@ -7517,8 +7570,13 @@ static int find_paths_to_edit(sd_bus *bus, char **names, char ***paths) {
 
                         if (arg_full)
                                 r = unit_file_create_copy(&lp, unit_name, path, &new_path, &tmp_path);
-                        else
-                                r = unit_file_create_new(&lp, unit_name, ".d/override.conf", &new_path, &tmp_path);
+                        else {
+                                r = strv_push_prepend(&unit_paths, strdup(path));
```

Comment:
leaks memory on failure...

please use strv_prepend()

Reply:
Swapped `strv_push_prepend` to `strv_consume_prepend`